tools/cli/internal/openapi/paths.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openapi
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/tufin/oasdiff/diff"
+)
+
+//go:generate mockgen -destination=../openapi/mock_paths.go -package=openapi github.com/mongodb/openapi/tools/cli/internal/openapi NoExtensionDiff
+
+const (
+	xgenSoaMigration = "x-xgen-soa-migration"
+	allowDocsDiff    = "allowDocsDiff"
+)
+
+// allOperationsHaveExtension checks if all the operations in the base pat have the given extension name.
+func allOperationsHaveExtension(basePath *openapi3.PathItem, basePathName, extensionName string) bool {
+	if basePath.Operations() == nil || len(basePath.Operations()) == 0 {
+		return false
+	}
+
+	if basePath.Get != nil {
+		if result := getOperationExtensionWithName(basePath.Get, extensionName); result == nil {
+			return false
+		}
+	}
+
+	if basePath.Put != nil {
+		if result := getOperationExtensionWithName(basePath.Put, extensionName); result == nil {
+			return false
+		}
+	}
+
+	if basePath.Post != nil {
+		if result := getOperationExtensionWithName(basePath.Post, extensionName); result == nil {
+			return false
+		}
+	}
+
+	if basePath.Patch != nil {
+		if result := getOperationExtensionWithName(basePath.Patch, extensionName); result == nil {
+			return false
+		}
+	}
+
+	if basePath.Delete != nil {
+		if result := getOperationExtensionWithName(basePath.Delete, extensionName); result == nil {
+			return false
+		}
+	}
+
+	log.Println("Detected x-xgen-soa-migration annotation in all operations for path: ", basePathName)
```

Comment:
my understanding is that this function can be used for all type of extension. Is this correct? if yes, you should update this log message to use the extension name in the input.

Reply:
thanks! updated